### PR TITLE
i18n: add zh-CN translations

### PR DESCRIPTION
**Pull request by chappjc**
This adds Simplified Chinese (PRC) translations, tag zh-CN, done by @DominicTing.

The first commit contains a few fixes for existing issues.

The second are the translated strings, for reviewers of just the translations.

The third are the generated localized_html files.

NOTE: A few status strings in client/webserver/site/src/js/locales.js remain untranslated.  Maybe we can keep this because they are reportedly difficult to get one word translations for.

### CODE REVIEW

**Comment by chappjc** [comment]
@DominicTing You mentioned you had some comments and resolutions for the TODOs?  Your review may still be pending because I cannot see any comments from you yet.

**Comment by chappjc** [comment]
Oh, I almost forgot I need to generate screenshots for you.  Will put those up today.

**Comment by chappjc** [comment]
**Set App Password dialog**

image
image
image

 
image
image
image

**Comment by chappjc** [comment]
**Add a "X" Wallet**

image
image

image
image

**Comment by chappjc** [comment]
**Add a DEX**

image
image
image
image


image
image
image
image
(couldn't get to markets page)

**Comment by chappjc** [comment]
Buy/Sell buttons

image

image

image

image

**Comment by chappjc** [comment]
Limit order

image

image

image

image

**Comment by chappjc** [comment]
Depth chart

image


image

**Comment by chappjc** [comment]
**Wallets** (simnet, just like all the other screen shots)

image

image

**Comment by chappjc** [comment]
**Cancel Order**

image

image
image

File: client/webserver/locales/pt-br.go [hunk 1]
```diff
@@ -1,40 +1,38 @@
...
+	"add a file":                     "adicionar um arquivo",
+	"Submit":                         "Enviar",
+	"Confirm Registration":           "Confirma Registro",
+	"app_pw_reg":                     "Informe sua senha do app para confirmar seu registro na DEX.",
+	"reg_confirm_submit":             `Quando vc enviar esse formulário, <span id="feeDisplay"></span> DCR será gasto de sua carteira decred para pagar a taxa de registro.`,
+	"provided_markets":               "Essa DEX provê os seguintes mercados:",
+	"base_header":                    "Base",
+	"quote_header":                   "Quote",
+	"lot_size_header":                "Tamanho do Lote",
+	"lot_size_headsup":               `All trades are in multiples of the lot size.`,
```
Comment (chappjc) [inline client/webserver/locales/pt-br.go:35]: @vctt94 Would you please pt-BR translation for this revised string?

**Comment by DominicTing** [comment]
> **添加“X”钱包**
> 
> 图片
> 图片
> 
> 图片
> 图片

1. add a decred wallet:"添加Decred钱包"
2. wallet setting: "钱包设置"
3. account name: "账户名称"
4. pre-size funding inputs:"预计投入资金额"
5.  rpc username:"RPC 用户名"
6.  rpc password:"RPC 密码"
7.  rpc address:" RPC 地址"
8.  tle certificate:"TLE 证书"
9. fallback fee rate:"撤回费率"
10. highest acceptable fee rate:"允许的最高费率"
"redeem confirmation target" What does this string mean?

**Comment by DominicTing** [comment]
> **Set App Password dialog**
> 
> image
> image
> image
> image
> 
> image
> image
> image
> image

1. "don't forget...":"不要忘记备份你的应用程序种子"
2. "a new app...":"已创建新的应用程序种子。请立刻在设置界面中进行备份。"

**Comment by DominicTing** [comment]
> **Add a DEX**
> 
> image
> image
> image
> image
> 
> image
> image
> image
> image

1. this dex provides the following markets:"当前DEX提供以下市场："
2. all trades are in multiples of the lot size:"所有交易都是手数的倍数。"

revise:
1. lot size:"手数"
2. immature:"未成熟"

**Comment by DominicTing** [comment]
> 限价单
> 
> 图片
> 
> 图片
> 
> 图片
> 
> 图片
> 
> 图片
> 
> 图片

1. lots:"批"

revise：
1. Max Buy:"最多买入"
2. rate step:"兑换"
3. verify buy order:"检查买单"
4.  you are submitting an order to:"您正在提交一下订单"
5.  for a total of:"共计"
6.  rate:"汇率"

**Comment by DominicTing** [comment]
> 深度图
> 
> 图片
> 
> 图片

1.sells:"卖"
2.volume:"量"

**Comment by DominicTing** [comment]
> **钱包**（simnet，就像所有其他屏幕截图一样）
> 
> 图片
> 
> 图片

1. ready:"准备就绪"
2. no wallet:"未连接钱包"
3. create a Bitcoin Cash wallet:"创建一个Bitcoin Cash钱包"
4. create a Litcoin wallet:"创建一个Litcoin钱包"

revise
1.Asset:"列表"
2.Balance:"余额"
3.Actions:"操作"

**Comment by DominicTing** [comment]
> **取消订单**
> 
> 图片
> 
> 图片
> 图片

1. api error:"接口错误"

revise:
1. Cancel Order:"取消订单"
2. Enter your...10DCR.:"输入密码以取消订单未成交的10DCR。"

File: client/core/core.go [hunk 2]
```diff
@@ -1126,9 +1125,9 @@ func New(cfg *Config) (*Core, error) {
 			}
 			langs = append(langs, tag)
 		}
-		// fmt.Println("defined langs:", langs)
 		matcher := language.NewMatcher(langs)
-		tag, _, conf := matcher.Match(acceptLangs...)
+		_, idx, conf := matcher.Match(acceptLang) // use index because tag may end up as something hyper specific like zh-Hans-u-rg-cnzzzz
+		tag := langs[idx]
```
Comment (chappjc) [inline client/core/core.go:1130]: We needed to be using the index from the matcher, like we were doing in webserver.go for the reason described in the comment.

File: client/webserver/locales/en-us.go [hunk 3]
```diff
@@ -56,6 +56,7 @@ var EnUS = map[string]string{
 	"price":                          "price",
 	"volume":                         "volume",
 	"buys":                           "buys",
+	"sells":                          "sells",
```
Comment (chappjc) [inline client/webserver/locales/en-us.go:59]: For the depth chart header, there is "buys" and "sells" but only "buys" had a translation.

File: client/webserver/locales/pt-br.go [hunk 4]
```diff
@@ -1,40 +1,38 @@
...
+	"add a file":                     "adicionar um arquivo",
+	"Submit":                         "Enviar",
+	"Confirm Registration":           "Confirma Registro",
+	"app_pw_reg":                     "Informe sua senha do app para confirmar seu registro na DEX.",
+	"reg_confirm_submit":             `Quando vc enviar esse formulário, <span id="feeDisplay"></span> DCR será gasto de sua carteira decred para pagar a taxa de registro.`,
+	"provided_markets":               "Essa DEX provê os seguintes mercados:",
+	"base_header":                    "Base",
+	"quote_header":                   "Quote",
+	"lot_size_header":                "Tamanho do Lote",
+	"lot_size_headsup":               `Todas as trocas são múltiplas do tamanho do lote.`,
```
Comment (chappjc) [inline client/webserver/locales/pt-br.go:35]: `lot_size_headsup` changed with the addition of the market table.

**Comment by chappjc** [comment]
I think this is translated as much as we can hope for now.

There are changes unrelated to the ZH work in commit https://github.com/decred/dcrdex/pull/1207/commits/c14661737bc51817c8c4e835e8b8ca463e4a02f3 that should be reviewed as well.  This can be put in another PR too if needed.

I haven't retested since today's pushes.

File: client/webserver/locales/en-us.go (hunk 3 above)
Comment (DominicTing) [reply client/webserver/locales/en-us.go:59]: sells: "卖"

File: client/webserver/webserver.go [hunk 5]
```diff
@@ -367,13 +367,14 @@ func (s *WebServer) buildTemplates(lang string) error {
 
 	if match == "" {
 		// Try to identify candate languages.
```
Comment (buck54321) [inline client/webserver/webserver.go:369]: candate

**Review by buck54321 — APPROVED** [review]

**Pull request merged**